multiversx_sdk_cli/cli_shared.py

```diff
@@ -369,9 +385,15 @@ def prepare_account(args: Any):
         except Exception as e:
             raise LedgerError(str(e))
     elif args.sender:
-        return load_wallet_by_alias(alias=args.sender, hrp=hrp)
+        acc = load_wallet_by_alias(alias=args.sender, hrp=hrp)
+        if _options_set_for_hash_signing(args):
+            acc.use_hash_signing = True
+        return acc
     else:
-        return load_default_wallet(hrp=hrp)
+        acc = load_default_wallet(hrp=hrp)
+        if _options_set_for_hash_signing(args):
+            acc.use_hash_signing = True
```

Comment:
Can this be refactored to be written only once?

Oh, it's not on all flows, sorry (not on `LedgerAccount`).